src/ceph-volume/ceph_volume/util/system.py

```diff
@@ -33,21 +33,31 @@ def generate_uuid():
 
 def which(executable):
     """find the location of an executable"""
-    locations = (
+    def _get_path(executable, locations):
+        for location in locations:
+            executable_path = os.path.join(location, executable)
+            if os.path.exists(executable_path) and os.path.isfile(executable_path):
+                return executable_path
+
+    path = os.getenv('PATH', '')
+    path_locations = path.split(':')
+    exec_in_path = _get_path(executable, path_locations)
+    if exec_in_path:
+        return exec_in_path
+    mlogger.warning('Absolute path not found for executable: %s', executable)
+    mlogger.warning('Current $PATH is: %s', path)
```

Comment:
Mind changing the format strings to use `'some string'.format()`?

Reply:
Sure